demisto_sdk/commands/common/tools.py

```diff
@@ -1010,98 +1009,106 @@ def find_type_by_path(path: str = '') -> Optional[FileType]:
             return FileType.INCIDENT_TYPE
 
         # 'regex' key can be found in new reputations files while 'reputations' key is for the old reputations
         # located in reputations.json file.
-        if 'regex' in _dict or 'reputations' in _dict:
+        if 'regex' in file_data or 'reputations' in file_data:
             return FileType.REPUTATION
 
-        if 'brandName' in _dict and 'transformer' in _dict:
+        if 'brandName' in file_data and 'transformer' in file_data:
             return FileType.OLD_CLASSIFIER
 
-        if ('transformer' in _dict and 'keyTypeMap' in _dict) or 'mapping' in _dict:
-            if _dict.get('type') and _dict.get('type') == 'classification':
+        if ('transformer' in file_data and 'keyTypeMap' in file_data) or 'mapping' in file_data:
+            if file_data.get('type') and file_data.get('type') == 'classification':
                 return FileType.CLASSIFIER
-            elif _dict.get('type') and 'mapping' in _dict.get('type'):
+            elif file_data.get('type') and 'mapping' in file_data.get('type'):
                 return FileType.MAPPER
             return None
 
-        if 'canvasContextConnections' in _dict:
+        if 'canvasContextConnections' in file_data:
             return FileType.CONNECTION
 
-        if 'layout' in _dict or 'kind' in _dict:
-            if 'kind' in _dict or 'typeId' in _dict:
+        if 'layout' in file_data or 'kind' in file_data:
+            if 'kind' in file_data or 'typeId' in file_data:
                 return FileType.LAYOUT
 
             return FileType.DASHBOARD
 
-        if 'group' in _dict and LAYOUT_CONTAINER_FIELDS.intersection(_dict):
+        if 'group' in file_data and LAYOUT_CONTAINER_FIELDS.intersection(file_data):
             return FileType.LAYOUTS_CONTAINER
 
         # When using it for all files validation- sometimes 'id' can be integer
-        if 'id' in _dict:
-            if isinstance(_dict['id'], str):
-                _id = _dict['id'].lower()
+        if 'id' in file_data:
+            if isinstance(file_data['id'], str):
+                _id = file_data['id'].lower()
                 if _id.startswith('incident'):
                     return FileType.INCIDENT_FIELD
                 if _id.startswith('indicator'):
                     return FileType.INDICATOR_FIELD
             else:
                 print(f'The file {path} could not be recognized, please update the "id" to be a string')
+        return None
 
-    return None
+@lru_cache()
+def find_type(path: str = '', ignore_sub_categories: bool = False):
+    """
+    returns the content file type
+
+    Arguments:
+        path - a path to the file
+
+    Returns:
+        string representing the content file type
+    """
+    type_by_path = find_type_by_path(path)
+    if type_by_path:
+        return type_by_path
+
+    return find_file_by_content(path, ignore_sub_categories)
```

Comment:
I don't like to split this into 2 separate ones.
This method has one function and it should be kept in the same place to ease the process of following it.

This adds steps to debugging which are not necessary

Reply:
The improvement is already inside.
The only thing that changes here is the Download in the content repo.